src/App/integrationTests/collectRecords/benthicLit/App.createOnlineBenthicLit.test.js

```diff
@@ -1,48 +1,117 @@
 
-const saveNewRecord = async () => {
-  userEvent.selectOptions(await screen.findByLabelText('Site'), '1')
-  userEvent.selectOptions(screen.getByLabelText('Management'), '2')
-  userEvent.type(screen.getByLabelText('Depth'), '10000')
-  userEvent.type(screen.getByLabelText('Sample Date'), '2021-04-21')
-  userEvent.type(screen.getByLabelText('Sample Time'), '12:34')
-
-  userEvent.type(screen.getByLabelText('Transect Number'), '56')
-  userEvent.type(screen.getByLabelText('Label'), 'some label')
-  userEvent.type(screen.getByLabelText('Transect Length Surveyed'), '2')
-  userEvent.selectOptions(
-    screen.getByLabelText('Reef Slope'),
-    'c04bcf7e-2d5a-48d3-817a-5eb2a213b6fa',
+const saveNewRecord = async (user) => {
+  const siteInput = await screen.findByLabelText('Site')
+  const managementRegimeInput = screen.getByLabelText('Management')
+  const depthInput = screen.getByLabelText('Depth')
+  const sampleDateInput = screen.getByLabelText('Sample Date')
+  const sampleTimeInput = screen.getByLabelText('Sample Time')
+  const transectNumberInput = screen.getByLabelText('Transect Number')
+  const labelInput = screen.getByLabelText('Label')
+  const transectLengthInput = screen.getByLabelText('Transect Length Surveyed')
+  const reefSlopeInput = screen.getByLabelText('Reef Slope')
+  const visibilityInput = screen.getByLabelText('Visibility')
+  const currentInput = screen.getByLabelText('Current')
+  const relativeDepthInput = screen.getByLabelText('Relative Depth')
+  const tideInput = screen.getByLabelText('Tide')
+  const notesInput = screen.getByLabelText('Notes')
+
+  await user.selectOptions(siteInput, '1')
+  await waitFor(() =>
+    expect(within(siteInput).getByRole('option', { name: 'Site A' }).selected).toBe(true),
+  )
+
+  await user.selectOptions(managementRegimeInput, '2')
+  await waitFor(() =>
+    expect(
+      within(managementRegimeInput).getByRole('option', {
+        name: 'Management Regimes B [Management Regimes 2]',
+      }).selected,
+    ).toBe(true),
   )
-  userEvent.selectOptions(
-    screen.getByLabelText('Visibility'),
-    'a3ba3f14-330d-47ee-9763-bc32d37d03a5',
+
+  await user.type(depthInput, '10000')
+  expect(depthInput).toHaveValue(10000)
+
+  await user.type(sampleDateInput, '2021-04-21')
+  expect(sampleDateInput).toHaveValue('2021-04-21')
+
+  await user.type(sampleTimeInput, '12:34')
+  expect(sampleTimeInput).toHaveValue('12:34')
+
+  await user.type(transectNumberInput, '56')
+  expect(transectNumberInput).toHaveValue(56)
+
+  await user.type(labelInput, 'some label')
+  expect(labelInput).toHaveValue('some label')
+
+  await user.type(transectLengthInput, '2')
+  expect(transectLengthInput).toHaveValue(2)
+
+  await user.selectOptions(reefSlopeInput, 'c04bcf7e-2d5a-48d3-817a-5eb2a213b6fa')
+  await waitFor(() =>
+    expect(
+      within(reefSlopeInput).getByRole('option', {
+        name: 'flat',
+      }).selected,
+    ).toBe(true),
```

Comment:
Example of having to assert that a form value is actually changed after a user event. (sometimes what is failing is an assertion further down, like something expecting a button to be enabled)